src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int CAPACITY = 10;
+    private static final double COEFFICIENT = 0.5;
+    private static final int NUMBER_ONE = 1;
+    private T[] array;
+    private int size;
 
+    public MateStack() {
+        array = (T[]) new Object[CAPACITY];
+        size = 0;
+    }
+
+    public void push(T value) {
+        if (size >= array.length) {
+            array = resize(array);
+        }
+        array[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        return array[size - NUMBER_ONE];
     }
 
     public T pop() {
-        return null;
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+        T currentValue = array[size - NUMBER_ONE];
+        size--;
+        array[size] = null;
```

Comment:
```suggestion
        T currentValue = array[size - NUMBER_ONE];
        size--;
        array[size - 1] = null;
```
we return size -  1 and after that we said that size will equal null, maybe size - 1 will equal null, I'm not sure about that

Reply:
In this logic i decrease size and than use this size (which already - 1) and this work correctly. But I aggre with you and I changed line 37 and 38 places and use for understanding array[size - ONE] = null

Reply:
understood )